packages/data/tests/unit/serializers/dimensions/elide-test.ts

```diff
@@ -0,0 +1,69 @@
+import { module, test } from 'qunit';
+import { setupTest } from 'ember-qunit';
+import { QueryStatus, AsyncQueryResponse, QueryResultType } from 'navi-data/adapters/facts/interface';
+import { DimensionColumn } from 'navi-data/adapters/dimensions/interface';
+import DimensionMetadataModel from 'navi-data/models/metadata/dimension';
+import NaviDimensionModel from 'navi-data/models/navi-dimension';
+import { TestContext } from 'ember-test-helpers';
+
+module('Unit | Serializer | Dimensions | Elide', function(hooks) {
+  setupTest(hooks);
+
+  test('normalize', function(this: TestContext, assert) {
+    assert.expect(3);
+    const serializer = this.owner.lookup('serializer:dimensions/elide');
+
+    const payload: AsyncQueryResponse = {
+      asyncQuery: {
+        edges: [
+          {
+            node: {
+              id: 'c7d2fe70-b63f-11ea-b45b-bf754c72eca6',
+              query: '"{ "query": "{ tableA { edges { node { dimension1 } } } } " }',
+              status: QueryStatus.COMPLETE,
+              result: {
+                contentLength: 129,
+                httpStatus: 200,
+                resultType: QueryResultType.EMBEDDED,
+                responseBody:
+                  '{"data":{"tableA":{"edges":[{"node":{"dimension1":"foo"}},{"node":{"dimension1":"bar"}},{"node":{"dimension1":"baz"}}]}}}'
+              }
+            }
+          }
+        ]
+      }
+    };
+    const dimColumn: DimensionColumn = {
+      columnMetadata: DimensionMetadataModel.create(this.owner.ownerInjection(), {
+        id: 'dimension1',
+        source: 'elideTwo',
+        tableId: 'tableA'
+      }),
+      parameters: {
+        baddabing: 'baddaboom'
+      }
+    };
+    assert.deepEqual(serializer.normalize(dimColumn), [], 'Empty array is returned for an undefined payload');
+
+    const noTableColumn = Object.assign({}, dimColumn, {
+      columnMetadata: DimensionMetadataModel.create(this.owner.ownerInjection(), {
+        id: 'dimension1',
+        source: 'elideTwo'
+      })
+    });
+    assert.deepEqual(
+      serializer.normalize(noTableColumn, payload),
+      [],
+      'Empty array is returned for column with no table id defined'
```

Comment:
Is this a real case for Elide?

Reply:
If so, should we throw an error?

Reply:
Not sure it is, but the tableId is an optional property on the metadata model.